src/components/Client/index.js

```diff
@@ -23,6 +23,17 @@ import { statusChanged } from '../../actions/status';
 class Client extends Component {
     constructor(props) {
         super(props);
+
+        /**
+         * I wonder if there is a way to have the Client View component
+         * know less about the socket.
+         *
```

Comment:
I wasn't certain how that would work, since it wouldn't actually be a Component with a render method. I'm sure there's sure there's a way to integrate it more tightly with state. 

However, I figured that there are probably cases where you want to integrate with a non-React component that just exposes an API for sending in callbacks. This Socket class would totally be reusable in a non-React environment. E.g, I could easily fold it back into the rudimentary client that exists in the server project, and it's actually attractive to me to do that, as a way of demonstrating code reuse.